File: app/src/main/java/rocks/poopjournal/todont/Fragments/HabitsFragment.java
package rocks.poopjournal.todont.Fragments;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.content.SharedPreferences;
import android.database.sqlite.SQLiteException;
import android.graphics.Color;
import android.os.Bundle;
import android.util.Log;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RelativeLayout;
import android.widget.Spinner;
import android.widget.SpinnerAdapter;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.bottomsheet.BottomSheetDialog;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

import rocks.poopjournal.todont.Db_Controller;
import rocks.poopjournal.todont.Adapters.HabitsAdapter;
import rocks.poopjournal.todont.Helper;
import rocks.poopjournal.todont.MainActivity;
import rocks.poopjournal.todont.R;


public class HabitsFragment extends Fragment {
    RecyclerView rv;
    ArrayList<String> gettingtasks = new ArrayList<>();
    ArrayList<String> gettingcatagory = new ArrayList<>();
    FloatingActionButton floatingActionButton;
    Date c = Calendar.getInstance().getTime();
    String catagoryselected;
    Db_Controller db;
    HabitsAdapter adapter;
    TextView tv1,tv2;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view= inflater.inflate(R.layout.fragment_habits, container, false);
        Helper.SelectedButtonOfTodayTab=0;
        rv = view.findViewById(R.id.rv);
        db = new Db_Controller(getActivity(), "", null, 2);
        db.show_labels();
        db.show_habits_data();
        db.getNightMode();
        tv1=view.findViewById(R.id.a);
        tv2=view.findViewById(R.id.b);
        floatingActionButton = view.findViewById(R.id.floatingbtn);
        floatingActionButton.bringToFront();
        floatingActionButton.setVisibility(View.VISIBLE);
        if(Helper.habitsdata.size()!=0){
            tv1.setVisibility(View.INVISIBLE);
            tv2.setVisibility(View.INVISIBLE);
        }
        else if(Helper.habitsdata.size()==0){
            tv1.setVisibility(View.VISIBLE);
            tv2.setVisibility(View.VISIBLE);
        }

        floatingActionButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                final SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd", Locale.getDefault());
                final String formattedDate = df.format(c);
                final Dialog d = new Dialog(getActivity());
                d.requestWindowFeature(Window.FEATURE_NO_TITLE);
                d.setContentView(R.layout.dialogbox_floatingbutton);
                Button btndone=d.findViewById(R.id.btndone);
                WindowManager.LayoutParams lp = d.getWindow().getAttributes();
                lp.dimAmount=0.9f;
                lp.gravity = Gravity.BOTTOM;
                Window window = d.getWindow();
                window.setLayout(WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.WRAP_CONTENT);
                d.getWindow().setAttributes(lp);
                d.getWindow().addFlags(WindowManager.LayoutParams.FLAG_BLUR_BEHIND);

                final Spinner spinner = d.findViewById(R.id.spinner);
                final TextView txt=d.findViewById(R.id.txt);
                if(Helper.labels_array.size()==0){
                    txt.setVisibility(View.VISIBLE);
                    spinner.setVisibility(View.INVISIBLE);
                }
                else{
                    txt.setVisibility(View.INVISIBLE);
                    spinner.setVisibility(View.VISIBLE);
                }


                ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(getActivity(),android.R.layout.simple_spinner_item,Helper.labels_array);
                arrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
                spinner.setAdapter(arrayAdapter);


                spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
                    @Override
                    public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
                        catagoryselected = adapterView.getItemAtPosition(i).toString();
//                        if(Helper.isnightmodeon.equals("no")){
                            ((TextView) adapterView.getChildAt(0)).setTextColor(getResources().getColor(R.color.textcolor));
//
//                        }
//                        else if(Helper.isnightmodeon.equals("yes")){
//                            ((TextView) adapterView.getChildAt(0)).setTextColor(Color.WHITE);
//
//                        }
                    }

                    @Override
                    public void onNothingSelected(AdapterView<?> adapterView) {

                    }
                });

                Button saveTaskButton = d.findViewById(R.id.saveTaskButton);
                final EditText habit = d.findViewById(R.id.habit);
                final EditText detail = d.findViewById(R.id.detail);

                saveTaskButton.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {

                        String habit_text = habit.getText().toString();
                        String detail_text = detail.getText().toString();
                        //String formattedDate = df.format(c);
                        try {

                        } catch (SQLiteException e) {
                        }
                        if(habit_text.equals("")){

                        }
                        db.insert_habits(Helper.habitsdata.size(),formattedDate, habit_text, detail_text, catagoryselected);
                        db.show_habits_data();

                        Helper.SelectedButtonOfTodayTab=0;
                        Intent i=new Intent(getActivity(), MainActivity.class);
                        startActivity(i);
                        getActivity().overridePendingTransition(0,0);
                        d.dismiss();
                    }

                });
                spinner.setAdapter(arrayAdapter);
                d.show();


            }
        });


//        floatingActionButton.setOnClickListener(new View.OnClickListener() {
//            @SuppressLint("ResourceAsColor")
//            @Override
//            public void onClick(View view) {
//                final SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd", Locale.getDefault());
//                final String formattedDate = df.format(c);
//                View bottomsheetview = null;
//                final BottomSheetDialog bottomSheetDialog = new BottomSheetDialog(getActivity(),
//                        R.style.BottomSheetDialogTheme);
//                bottomsheetview = LayoutInflater.from(getActivity().getApplicationContext()).
//                            inflate(R.layout.layout_bottom_sheet,
//                                    (RelativeLayout) view.findViewById(R.id.bottomsheetContainer));
////                if(Helper.isnightmodeon.equals("no")){
////                    bottomsheetview = LayoutInflater.from(getActivity().getApplicationContext()).
////                            inflate(R.layout.layout_bottom_sheet,
////                                    (RelativeLayout) view.findViewById(R.id.bottomsheetContainer));
////                }
////                else if(Helper.isnightmodeon.equals("yes")){
////                    bottomsheetview = LayoutInflater.from(getActivity().getApplicationContext()).
////                            inflate(R.layout.layout_bottom_sheet_nightmode,
////                                    (RelativeLayout) view.findViewById(R.id.bottomsheetContainer));
////                }
//                final Spinner spinner = bottomsheetview.findViewById(R.id.spinner);
//                final TextView txt=bottomsheetview.findViewById(R.id.txt);
//                if(Helper.labels_array.size()==0){
//                    txt.setVisibility(View.VISIBLE);
//                    spinner.setVisibility(View.INVISIBLE);
//                }
//                else{
//                    txt.setVisibility(View.INVISIBLE);
//                    spinner.setVisibility(View.VISIBLE);
//                }
//
//
//                ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(getActivity(),android.R.layout.simple_spinner_item,Helper.labels_array);
//                arrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
//                spinner.setAdapter(arrayAdapter);
//
//
//                spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
//                    @Override
//                    public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
//                        catagoryselected = adapterView.getItemAtPosition(i).toString();
//                        if(Helper.isnightmodeon.equals("no")){
//                            ((TextView) adapterView.getChildAt(0)).setTextColor(Color.BLACK);
//
//                        }
//                        else if(Helper.isnightmodeon.equals("yes")){
//                            ((TextView) adapterView.getChildAt(0)).setTextColor(Color.WHITE);
//
//                        }
//                    }
//
//                    @Override
//                    public void onNothingSelected(AdapterView<?> adapterView) {
//
//                    }
//                });
//
//                Button saveTaskButton = bottomsheetview.findViewById(R.id.saveTaskButton);
//                final EditText habit = bottomsheetview.findViewById(R.id.habit);
//                final EditText detail = bottomsheetview.findViewById(R.id.detail);
//
//                saveTaskButton.setOnClickListener(new View.OnClickListener() {
//                    @Override
//                    public void onClick(View view) {
//
//                        String habit_text = habit.getText().toString();
//                        String detail_text = detail.getText().toString();
//                        //String formattedDate = df.format(c);
//                        try {
//
//                        } catch (SQLiteException e) {
//                        }
//                        if(habit_text.equals("")){
//
//                        }
//                        db.insert_habits(Helper.habitsdata.size(),formattedDate, habit_text, detail_text, catagoryselected);
//                        db.show_habits_data();
//
//                        Helper.SelectedButtonOfTodayTab=0;
//                        Intent i=new Intent(getActivity(), MainActivity.class);
//                        startActivity(i);
//                        getActivity().overridePendingTransition(0,0);
//                        bottomSheetDialog.dismiss();
//                    }
//
//                });
//                spinner.setAdapter((SpinnerAdapter) arrayAdapter);
//                bottomSheetDialog.setContentView(bottomsheetview);
//                bottomSheetDialog.show();
//
//
//            }
//        });
        setDataInList();
        return view;

    }
    public void setDataInList() {
        for (int i = 0; i < Helper.habitsdata.size(); i++) {
            gettingtasks.add(Helper.habitsdata.get(i)[2]);
            gettingcatagory.add(Helper.habitsdata.get(i)[4]);
        }
       rv.setLayoutManager(new LinearLayoutManager(getActivity()));
        new ItemTouchHelper(itemtouchhelper).attachToRecyclerView(rv);
        adapter= new HabitsAdapter(getActivity(), HabitsFragment.this,db,gettingtasks,gettingcatagory);
        rv.setAdapter(adapter);
        rv.setLayoutManager(new LinearLayoutManager(getActivity()));
    }
    ItemTouchHelper.SimpleCallback itemtouchhelper = new ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.RIGHT) {
        @Override
        public boolean onMove(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull RecyclerView.ViewHolder target) {
            return false;
        }

        @Override
        public void onSwiped(@NonNull final RecyclerView.ViewHolder viewHolder, int direction) {
            if (direction == 8) {
                AlertDialog.Builder builder1 = new AlertDialog.Builder(getContext());
                builder1.setMessage("Do you really want to delete this?");
                builder1.setCancelable(true);

                builder1.setPositiveButton(
                        "Yes",
                        new DialogInterface.OnClickListener() {
                            public void onClick(DialogInterface dialog, int id) {
                                int i = viewHolder.getAdapterPosition();
                                db.delete_habits(i);
                                for (int j = i + 1; j < Helper.habitsdata.size(); j++) {
                                    db.updateHabitsIdsAfterDeletion(Integer.parseInt(Helper.habitsdata.get(j)[0]));
                                }
                                Helper.SelectedButtonOfTodayTab=0;
                                Intent intent = new Intent(getActivity(), MainActivity.class);
                                startActivity(intent);
                                getActivity().overridePendingTransition(0, 0);
                                dialog.cancel();
                            }
                        });

                builder1.setNegativeButton(
                        "No",
                        new DialogInterface.OnClickListener() {
                            public void onClick(DialogInterface dialog, int id) {
                                Helper.SelectedButtonOfTodayTab=0;
                                Intent i = new Intent(getActivity(), MainActivity.class);
                                startActivity(i);
                                getActivity().overridePendingTransition(0, 0);
                                dialog.cancel();
                            }
                        });

                AlertDialog alert11 = builder1.create();
                alert11.show();

            }

        }
    };
}

File: app/src/main/java/rocks/poopjournal/todont/fragments/HabitsFragment.kt
package rocks.poopjournal.todont.fragments

import android.app.Activity
import android.app.AlertDialog
import android.app.Dialog
import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.net.Uri
import android.os.Bundle
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.Window
import android.view.WindowManager
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.TextView
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.google.android.material.floatingactionbutton.FloatingActionButton
import jp.wasabeef.glide.transformations.BlurTransformation
import rocks.poopjournal.todont.adapters.HabitsAdapter
import rocks.poopjournal.todont.Helper
import rocks.poopjournal.todont.R
import rocks.poopjournal.todont.databinding.DialogboxAddNewHabitBinding
import rocks.poopjournal.todont.model.Habit
import rocks.poopjournal.todont.model.Label
import rocks.poopjournal.todont.showcaseview.ShowcaseViewBuilder
import rocks.poopjournal.todont.showcaseview.ShowcaseViewBuilder.Companion.init
import rocks.poopjournal.todont.utils.Constants
import rocks.poopjournal.todont.utils.DatabaseUtils
import rocks.poopjournal.todont.utils.SharedPrefUtils
import smartdevelop.ir.eram.showcaseviewlib.GuideView
import smartdevelop.ir.eram.showcaseviewlib.config.DismissType
import smartdevelop.ir.eram.showcaseviewlib.config.Gravity
import smartdevelop.ir.eram.showcaseviewlib.config.PointerType
import java.io.File
import java.io.FileOutputStream
import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Date
import java.util.Locale

class HabitsFragment() : Fragment() {
    private var rv: RecyclerView? = null
    private var habitsList = ArrayList<Habit>()
    private var date: Date = Calendar.getInstance().time
    private var selectedLabel: Label? = null
    private var dbHelper: DatabaseUtils? = null
    private var adapter: HabitsAdapter? = null
    private var tvNoHabits: TextView? = null
    private var showcaseViewBuilder: ShowcaseViewBuilder? = null
    private var prefUtils: SharedPrefUtils? = null


    private val PICK_IMAGE_REQUEST = 1001
    private var selectedImageUri: Uri? = null
    private var addNewHabit=false;

    private var dialogView: DialogboxAddNewHabitBinding? = null;


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.getBoolean("ARG_DATA")?.let {
            addNewHabit=it
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_habits, container, false)
        Helper.SelectedButtonOfTodayTab = 0
        rv = view.findViewById(R.id.rv)
        dbHelper = DatabaseUtils(requireContext())

        showcaseViewBuilder = init(requireActivity())
        prefUtils = SharedPrefUtils(requireActivity())
        //  tv1 = view.findViewById(R.id.a);
        tvNoHabits = view.findViewById(R.id.b)
        if(addNewHabit){
            addNewHabit()
            addNewHabit=false
        }

        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
    }


    override fun onResume() {
        super.onResume()
        setDataInList()
    }

    fun addNewHabit() {

        val labelCount = dbHelper?.getLabelsCount()
        if (labelCount == 0) {
            Toast.makeText(activity, R.string.please_add_a_label_first, Toast.LENGTH_LONG)
                .show()
        } else {
            val dateFormat = SimpleDateFormat(Constants.DATE_FORMAT, Locale.getDefault())
            val formattedDate = dateFormat.format(date)

            val dialog = Dialog(requireActivity())
            dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
            dialogView = DialogboxAddNewHabitBinding.inflate(layoutInflater)
            dialog.setContentView(dialogView!!.root)

            val lp = dialog.window!!.attributes
            lp.dimAmount = 0.9f
            dialog.window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
            val window = dialog.window
            window!!.setLayout(
                WindowManager.LayoutParams.MATCH_PARENT,
                WindowManager.LayoutParams.WRAP_CONTENT
            )
            dialog.window!!.attributes = lp

            if (labelCount == 0) {
                dialogView!!.txt.visibility = View.VISIBLE
                dialogView!!.spinner.visibility = View.GONE
            } else {
                dialogView!!.txt.visibility = View.GONE
                dialogView!!.spinner.visibility = View.VISIBLE
            }
            val labels = dbHelper?.getAllLabels()!!
            labels.add(0, Label(-1, "Select a label"))
            val arrayAdapter = ArrayAdapter(
                requireActivity(),
                android.R.layout.simple_spinner_item,
                labels
            )
            arrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
            dialogView!!.spinner.adapter = arrayAdapter


            dialogView!!.spinner.onItemSelectedListener =
                object : AdapterView.OnItemSelectedListener {
                    override fun onItemSelected(
                        adapterView: AdapterView<*>,
                        view: View,
                        i: Int,
                        l: Long
                    ) {
                        if (i != 0) {
                            selectedLabel = labels[i]
                            val typedValue = TypedValue()
                            requireContext().theme.resolveAttribute(
                                R.attr.colorOnBackground2,
                                typedValue,
                                true
                            )
                            (adapterView.getChildAt(0) as TextView).setTextColor(
                                typedValue.data
                            )
                        }

                    }

                    override fun onNothingSelected(adapterView: AdapterView<*>?) {
                    }
                }

            dialogView!!.btnSelectImg.setOnClickListener {
                openImagePicker()
            }


            dialogView!!.saveTaskButton.setOnClickListener {
                val habit_text = dialogView?.habit?.text.toString()
                val detail_text = dialogView?.detail?.text.toString()

                val imageUriString = selectedImageUri?.toString()

                dbHelper?.insertHabit(
                    Habit(
                        date = formattedDate,
                        name = habit_text,
                        description = detail_text,
                        countAvoided = 0,
                        countDone = 0,
                        labelId = selectedLabel?.labelId!!,
                        coverImageUri = imageUriString  // Add this line
                    )
                )

                // Reset image selection
                selectedImageUri = null
                dialogView?.frameSelectedImage?.visibility = View.GONE

                Helper.SelectedButtonOfTodayTab = 0
//                    val i = Intent(activity, MainActivity::class.java)
//                    startActivity(i)
                setDataInList()
                requireActivity().overridePendingTransition(0, 0)
                dialog.dismiss()
            }
            dialogView!!.spinner.adapter = arrayAdapter
            dialog.show()
        }

    }

    // Function to load image with blur effect
    private fun loadImageWithBlur(imageUri: Uri) {
        // Show both ImageViews
        dialogView?.frameSelectedImage?.visibility = View.VISIBLE

        // Load blurred image into background ImageView
        dialogView?.ivBlurredBackground?.let {
            Glide.with(requireContext())
                .load(imageUri)
                .apply(RequestOptions.bitmapTransform(BlurTransformation(25, 3)))
                .into(it)
        }

        // Load normal image into foreground ImageView
        dialogView?.ivSelectedImage?.let {
            Glide.with(requireContext())
                .load(imageUri)
                .into(it)
        }
    }

    // Add this function to handle image selection
    private fun openImagePicker() {
        val intent = Intent(Intent.ACTION_PICK).apply {
            type = "image/*"
            putExtra(Intent.EXTRA_MIME_TYPES, arrayOf("image/jpeg", "image/png"))
        }
        startActivityForResult(intent, PICK_IMAGE_REQUEST)
    }

    // Handle the result
    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == PICK_IMAGE_REQUEST && resultCode == Activity.RESULT_OK) {
            val inputUri = data?.data
            inputUri?.let { uri ->
                // Save the image to internal storage and get the new URI
                val savedImageUri = saveImageToInternalStorage(uri)
                savedImageUri?.let {
                    // Set the selectedImageUri to the URI of the saved image
                    selectedImageUri = it
                    // Load the image (e.g., with blur)
                    loadImageWithBlur(it)
                }
            }
        }
    }

    private fun saveImageToInternalStorage(uri: Uri): Uri? {
        return try {
            // Open an input stream from the original URI
            val inputStream = requireContext().contentResolver.openInputStream(uri)
            inputStream?.use { stream ->
                // Get the internal storage directory for the app
                val directory = File(requireContext().filesDir, "Cover_Images")
                if (!directory.exists()) {
                    directory.mkdirs() // Create the directory if it doesn't exist
                }

                // Create a file in the app's internal storage
                val fileName = "image_${System.currentTimeMillis()}.jpg"
                val file = File(directory, fileName)

                // Write the image data to the file
                val outputStream = FileOutputStream(file)
                outputStream.use { out ->
                    stream.copyTo(out)
                }

                // Return the URI of the saved file
                Uri.fromFile(file)
            }
        } catch (e: Exception) {
            e.printStackTrace()
            null
        }
    }


    private fun setDataInList() {
        dbHelper?.getAllHabits()?.let {
            habitsList = it
            if (habitsList.isNotEmpty()) {
                tvNoHabits?.visibility = View.INVISIBLE

                rv?.layoutManager = LinearLayoutManager(activity)
                ItemTouchHelper(itemtouchhelper).attachToRecyclerView(rv)
                adapter =
                    HabitsAdapter(
                        requireContext(),
                        dbHelper!!, habitsList
                    )

                rv!!.adapter = adapter
                rv!!.layoutManager = LinearLayoutManager(activity)

            } else {
                tvNoHabits?.visibility = View.VISIBLE
            }

        } ?: run { tvNoHabits?.setVisibility(View.VISIBLE) }

    }

    var itemtouchhelper: ItemTouchHelper.SimpleCallback =
        object : ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.RIGHT) {
            override fun onMove(
                recyclerView: RecyclerView,
                viewHolder: RecyclerView.ViewHolder,
                target: RecyclerView.ViewHolder
            ): Boolean {
                return false
            }

            override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
                if (direction == 8) {
                    val builder1 = AlertDialog.Builder(
                        context
                    )
                    builder1.setMessage(R.string.do_you_really_want_to_delete_this)
                    builder1.setCancelable(true)

                    builder1.setPositiveButton(
                        R.string.yes
                    ) { dialog, id ->
                        val i = viewHolder.adapterPosition
                        habitsList[i].coverImageUri?.let { it1 -> deleteImageFromInternalStorage(it1) }
                        dbHelper?.deleteHabit(habitsList[i].id)
                        dbHelper?.deleteAllHabitRecords(habitsList[i].id)
                        Helper.SelectedButtonOfTodayTab = 0
                        habitsList.removeAt(i)
                        adapter?.notifyItemRemoved(i)
                        adapter?.notifyItemRangeChanged(0, habitsList.size)
//                        val intent = Intent(activity, MainActivity::class.java)
//                        startActivity(intent)
                        activity!!.overridePendingTransition(0, 0)
                        dialog.cancel()
                    }

                    builder1.setNegativeButton(
                        R.string.no
                    ) { dialog, id ->
                        Helper.SelectedButtonOfTodayTab = 0
//                        val i = Intent(activity, MainActivity::class.java)
//                        startActivity(i)
                        adapter?.notifyDataSetChanged()
                        activity!!.overridePendingTransition(0, 0)
                        dialog.cancel()
                    }

                    val alert11 = builder1.create()
                    alert11.show()
                }
            }

            private fun deleteImageFromInternalStorage(uriString: String): Boolean {
                val uri = Uri.parse(uriString)
                return try {
                    // Convert the URI to a File object
                    val file = File(uri.path)
                    // Check if the file exists and delete it
                    if (file.exists()) {
                        file.delete()
                    } else {
                        false // File does not exist
                    }
                } catch (e: Exception) {
                    e.printStackTrace()
                    false
                }
            }
        }

    companion object {
        fun newInstance(someData: Boolean): HabitsFragment {
            val fragment = HabitsFragment()
            val args = Bundle()
            args.putBoolean("ARG_DATA", someData)
            fragment.arguments = args
            return fragment
        }


    }
}
